include/components/wandering_component.hpp

```diff
@@ -0,0 +1,23 @@
+#ifndef FILE_WANDERING_COMPONENT_HPP
+#define FILE_WANDERING_COMPONENT_HPP
+
+#include "brickengine/components/component_impl.hpp"
+#include <functional>
+
+class WanderingComponent : public ComponentImpl<WanderingComponent> {
+public:
+    using DeathFunction = std::function<void (int entity_id)>;
+    WanderingComponent(DeathFunction on_death);
+    static std::string getNameStatic();
+
+    // Data
+    double duration = 1;
+    double elapsed_time = 0;
+    double waited_for = 0;
+    double wait_duration = 10;
+    int random = 0;
```

Comment:
The random variable should have its purpose made more clear by making it a Direction. In the system you generate a number between 0 and 1 and then use it as a direction. If you generate the random, turn it into a Direction and then store and use it, the code would be way nicer to read.